project-metrics/metrics_service/metrics/base.py

```diff
@@ -0,0 +1,99 @@
+"""Base interface for metrics to implement."""
+
+import abc
+from typing import Sequence, Optional, Text
+import stringcase
+
+import models
+
+
+class Metric(object):
+  """Abstract base class for health metrics."""
+
+  __metaclass__ = abc.ABCMeta
+
+  def __init__(self):
+    self._result = None
+
+  def __str__(self) -> Text:
+    return '%s: %s' % (self.label, self.formatted_result)
+
+  @property
+  def name(self):
+    """Metric name to use as DB identifier when storing results."""
+    return self.__class__.__name__
+
+  @property
+  def label(self) -> Text:
+    """Label for the metric (ex. TestCoverageMetric -> "Test Coverage")."""
+    return stringcase.titlecase(
+        self.name[:-6] if self.name.endswith('Metric') else self.name)
```

Comment:
Being smart about the presentation layer usually doesn't end well, this might eventually come back to bite you ;)
(but don't feel like you have to "fix" this now)

Reply:
It does the trick for any new metrics we define out-of-the-box, and if someone is unhappy with the label it can be overridden :)